### PR TITLE
Initialize development docs

### PR DESCRIPTION
**Pull request by tms1337**
**What's the problem this PR addresses?**

Create a documentation explaining the inner workings for developers wanting to contribute or understand the project in depth.

**How did you fix it?**

Directions for “inner working” docs:
- explain Module, CommonModule, fractal structure
- explain ServerModule and some examples like core -> rest
- explain ClientModule
- explain ClientModule native
- explain packages/ folder and how the app is started

### CODE REVIEW

**Comment by tms1337** [comment]
@larixer Would you care to check the initial docs I submitted so I can get some feedback on the general direction?

**Comment by larixer** [comment]
@tms1337 Yep, very nice intro! I like it very much!

File: docs/development/Modules Fractal Structure.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
+# Introduction
+
+This section will introduce you to the basic concepts required in order to understand this repository's structure and how to use it. It the long run it will help you both contribute to this repository and use it for your own projects. 
+
+In this part we will cover only basci concepts but not implementational details we actually use in this repository. We will leave that for another section.
```
Comment (larixer) [inline docs/development/Modules Fractal Structure.md]: `basci` -> `basic`

**Comment by larixer** [comment]
@tms1337 Do you have plans to continue this PR further? Or should I merge this PR - if so, could you check my review comments and apply fixes?

**Comment by tms1337** [comment]
@larixer Yes I have plans to continue, just had a hectic time at work so didn't find time for this.
Hopefully within next week or two I will get back to it.

**Comment by tms1337** [comment]
@larixer I fixed the typos, so you can merge this now.

What is left is:
- explain ServerModule and some examples like core -> rest
- explain ClientModule
- explain ClientModule native
- explain packages/ folder and how the app is started

But I would suggest opening a separate PR for each of these so that we get something done at least.

Best,

**Comment by larixer** [comment]
Thank you so much @tms1337 !

**Pull request merged**